internal/providers/vsphere/pke/adapter/gorm_store.go

```diff
@@ -0,0 +1,466 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"emperror.dev/emperror"
+	"emperror.dev/errors"
+	"github.com/jinzhu/gorm"
+	"github.com/sirupsen/logrus"
+
+	"github.com/banzaicloud/pipeline/internal/cluster/clusteradapter/clustermodel"
+	sqlJson "github.com/banzaicloud/pipeline/internal/database/sql/json"
+	intPKE "github.com/banzaicloud/pipeline/internal/pke"
+	"github.com/banzaicloud/pipeline/internal/providers/vsphere/pke"
+	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
+)
+
+const (
+	ClustersTableName  = "vsphere_pke_clusters"
+	NodePoolsTableName = "vsphere_pke_node_pools"
+)
+
+type gormVspherePKEClusterStore struct {
+	db *gorm.DB
+}
+
+func NewClusterStore(db *gorm.DB) pke.ClusterStore {
+	return gormVspherePKEClusterStore{
+		db: db,
+	}
+}
+
+type rolesModel []string
+
+func (m *rolesModel) Scan(v interface{}) error {
+	return sqlJson.Scan(v, m)
+}
+
+func (m rolesModel) Value() (driver.Value, error) {
+	return sqlJson.Value(m)
+}
+
+type nodePoolModel struct {
+	gorm.Model
+
+	Autoscaling bool
+	ClusterID   uint `gorm:"unique_index:idx_vsphere_pke_np_cluster_id_name"`
+	CreatedBy   uint
+	Size        int
+	MaxSize     uint
+	MinSize     uint
+	VCPU        int `gorm:"column:vcpu"`
+	RamMB       int
```

Comment:
`Ram`, but it should be `RAM`: https://go.googlesource.com/lint/+/d0100b6bd8b389f0385611eb39152c4d7c3a7905/lint.go#788